source/fundamentals/versioned-api.txt

```diff
@@ -0,0 +1,128 @@
+
+This guide shows you how to specify the **Versioned API** when connecting to
+a MongoDB instance or replica set. You can use the Versioned API feature to
+force the server to run operations with behavior compatible with the
+specified **API version**. An API version defines the expected behavior of the
+operations it covers and the format of server responses. If you change to
+a different API version, the operations are not guaranteed to be
+compatible and the server responses are not guaranteed to be similar.
+
+When you use the Versioned API feature with an official MongoDB driver, you
+can update your driver or server without worrying about backward compatibility
+issues of the commands covered by the Versioned API.
+
+See the server manual page on `Versioned API <https://docs.mongodb.com/v5.0/reference/versioned-api/>`__
+for more information including a list of commands it covers.
+
+The following sections describe how you can enable Versioned API for
+your MongoDB client and the options that you can specify.
+
+Enable Versioned API on a MongoDB Client
+----------------------------------------
+
+To enable the Versioned API, you must specify an API version in the settings
+of your MongoDB client. Once you instantiate a ``MongoClient`` instance with
+a specified Versioned API, all commands you run with that client use that
+version of the Versioned API. If you need to run commands not covered by the
+Versioned API or with a different version of it, instantiate a separate client.
+
+The example below shows how you can instantiate a ``MongoClient`` that
+sets the Versioned API version and connects to a server by performing the
+following operations:
+
+- Construct a ``ServerApi`` instance using the ``ServerApi.Builder``
+  helper class.
+- Specify a Versioned API version using a constant from the
+  ``ServerApiVersion`` class.
+- Construct a ``MongoClientSettings`` instance using the
+  ``MongoClientSettings.Builder`` class.
+- Specify a server to connect to using a ``ServerAddress`` instance.
+- Instantiate a ``MongoClient`` using the ``MongoClients.create()`` method
+  and pass your ``MongoClientSettings`` instance as a parameter.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/VersionedApiExample.java
+   :start-after: start serverAPIVersion
+   :end-before: end serverAPIVersion
+   :language: java
+   :dedent:
+
+.. warning::
+
+   If you specify an API version and connect to a MongoDB server that does
+   not support Versioned API, your application may raise an exception when
+   executing a command on your MongoDB server. If you use
+   a ``MongoClient`` that specifies the API version to query a server that
+   does not support it, your query could fail with the following exception
+   message:
+
+   .. code-block:: none
+      :copyable: false
+
+      com.mongodb.MongoQueryException: Query failed with error code 9 and error message 'Failed to parse: {...} Unrecognized field 'apiVersion'.' on server
```

Comment:
I'm surprised this is the exception you get.  What operation did you try that threw this exception?

Reply:
Maybe try a few different operations, both reads and writes, and see what you get.

Reply:
Just want to make sure we're referring to the same thing. If you're using a server version older than 5.0 (doesn't support Versioned API) and run commands on a client, you'll receive an error that complains about the unrecognized field apiVersion.

For `find()`:
```
com.mongodb.MongoQueryException: Query failed with error code 9 and error message 'Failed to parse: {...} Unrecognized field 'apiVersion'.' on server
```

For insertOne()
```
com.mongodb.MongoCommandException: Command failed with error 40415 (Location40415): 'BSON field 'insert.apiVersion' is an unknown field.' on server localhost:27017. The full response is {"ok": 0.0, "errmsg": "BSON field 'insert.apiVersion' is an unknown field.", "code": 40415, "codeName": "Location40415"}
```
For aggregation:
```
com.mongodb.MongoCommandException: Command failed with error 9 (FailedToParse): 'unrecognized field 'apiVersion'' on server localhost:27017. The full response is {"ok": 0.0, "errmsg": "unrecognized field 'apiVersion'", "code": 9, "codeName": "FailedToParse"}
```

I'll update the message so it only mentions unrecognized field apiVersion unless there's some other error that you expected.